app/main.py

```diff
@@ -1 +1,50 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str):
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str):
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills = ["JavaScript", "HTML", "CSS"]
+
+    def create_awesome_web_page(self):
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django"]
+
+    def create_powerful_api(self):
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills = ["Java", "Android studio"]
+
+    def create_smooth_mobile_app(self):
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(BackendDeveloper, FrontendDeveloper):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills =\
+            BackendDeveloper(name).skills + \
+            FrontendDeveloper(name).skills
```

Comment:
with inheritance, you don't need to call `__init__`, try to implement this

Reply:
Sorry, dont't udertand you. What do you mean?

Reply:
you should use `__init__ ` your parents

Reply:
I understand your idea, but I can't imagine how to do it )

Reply:
what about method `extends()` ?)

Reply:
I still don’t understand how it will help ) For me, the whole problem lies in the fact that `skills `is a method of an instance and not a class. I don’t understand how to access the `skills `of an instance, which, for example, is in the `BackendDeveloper `class, without creating an instance there.

Reply:
Are lines 41 and 42 correct? Maybe I'm looking in the wrong place

Reply:
Is it ok? 
![изображение](https://user-images.githubusercontent.com/70214623/177040237-e748d25c-d65d-4e87-b5f8-bf357948646c.png)